tools/arxiv/scripts/collect_ai_papers_extended.py

```
@@ -0,0 +1,888 @@
+                for line in f:
+                    arxiv_id = line.strip()
+                    if arxiv_id:
+                        self.collected_ids.add(arxiv_id)
+            logger.info(f"Loaded {len(self.collected_ids)} existing ArXiv IDs")
+        
+        # Also try to load the JSON collection if it exists
+        json_files = list(self.output_dir.glob("arxiv_ai_collection_*.json"))
+        if json_files:
+            json_file = sorted(json_files)[-1]
+            logger.info(f"Loading existing collection from {json_file}")
+            with open(json_file, 'r') as f:
```

Ruff (0.12.2)

Unnecessary mode argument

Remove mode argument

(UP015)